src/test/java/com/blazemeter/jmeter/correlation/core/templates/TemplateTest.java

```diff
@@ -0,0 +1,609 @@
+
+package com.blazemeter.jmeter.correlation.core.templates;
+
+import com.blazemeter.jmeter.correlation.core.RulesGroup;
+import com.blazemeter.jmeter.correlation.core.templates.repository.TemplateProperties;
+import org.apache.logging.log4j.core.util.Assert;
+import org.junit.Test;
+
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.awt.image.BufferedImage.TYPE_INT_RGB;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class TemplateTest {
+
+    private static final String BUILD_DESCRIPTION = "Description test";
+    private static final String BUILD_COMPONENT = "Component";
+    private static final String BUILD_RESPONSE_FILTERS = "responseFilters";
+    private static final String BUILD_REPOSITORY_ID = "repositoryId";
+    private static final String BUILD_CHANGES = "changes";
+    private static final String BUILD_AUTHOR = "Author";
+    private static final String BUILD_URL = "URL";
+    private static final String TEMPLATE_TYPE = "correlation";
+    private static final String SIEBEL_TEMPLATE_REFERENCE_NAME = "siebel";
+    private static final String TEMPLATE_VERSION_ONE = "1.0";
+
+
+    @Test
+    public void shouldReturnTrueWhenBuildersAreTheSameObject() {
+        Template.Builder builder = new Template.Builder();
+        builder.withId(SIEBEL_TEMPLATE_REFERENCE_NAME).withVersion(TEMPLATE_VERSION_ONE);
+        assertTrue(builder.equals(builder));
+    }
+
+    @Test
+    public void shouldReturnFalseWhenBuildersIsCompareWithObjectOfDifferentClass() {
+        Template.Builder builder = new Template.Builder();
+        builder.withId(SIEBEL_TEMPLATE_REFERENCE_NAME).withVersion(TEMPLATE_VERSION_ONE);
+        assertFalse(builder.equals("aa"));
+    }
+
+    @Test
+    public void shouldReturnAStringWhenToString() {
+        Template.Builder builder = new Template.Builder();
+        builder.withId(SIEBEL_TEMPLATE_REFERENCE_NAME)
```

Comment:
I've observed that the builder is extensively utilized across multiple tests, often with similar configurations. To enhance both maintainability and readability, I suggest defining a common template builder attribute, something like this maybe:

```java
private Template.Builder siebelTemplateBuilder = new Template.Builder()
        .withId(SIEBEL_TEMPLATE_REFERENCE_NAME)
        .withVersion(TEMPLATE_VERSION_ONE)
        .withDescription(BUILD_DESCRIPTION)
        .withGroups(new ArrayList<>());
        // Include additional standard configurations

```
This approach can then be applied in relevant tests, streamlining the setup process and reducing redundancy 😄 .